spec/values/calculation/max.hrx

```diff
@@ -0,0 +1,313 @@
+<===> README.md
+Most of the same behavior tested for `calc()` applies to `max()`, but for
+terseness' sake isn't tested explicitly.
+
+<===>
+================================================================================
+<===> error/syntax/no_args/input.scss
+a {b: max()}
+
+<===> error/syntax/no_args/error
+Error: At least one argument must be passed.
+  ,
+1 | a {b: max()}
+  |       ^^^^^
+  '
+  input.scss 1:7  root stylesheet
+
+<===>
+================================================================================
+<===> error/syntax/invalid_arg/input.scss
+a {b: max(c)}
+
+<===> error/syntax/invalid_arg/error
+Error: c is not a number.
+  ,
+1 | a {b: max(c)}
+  |       ^^^^^^
+  '
+  input.scss 1:7  root stylesheet
+
+<===>
+================================================================================
+<===> error/known_incompatible/first/input.scss
+a {b: max(1s, 2px)}
+
+<===> error/known_incompatible/first/error
+Error: 1s and 2px are incompatible.
+  ,
+1 | a {b: max(1s, 2px)}
+  |           ^^ 1s
+  |               === 2px
+  '
+  input.scss 1:11  root stylesheet
+
+<===>
+================================================================================
+<===> error/known_incompatible/second/input.scss
+a {b: max(1px, 2s)}
+
+<===> error/known_incompatible/second/error
+Error: 1px and 2s are incompatible.
+  ,
+1 | a {b: max(1px, 2s)}
+  |           ^^^ 1px
+  |                == 2s
+  '
+  input.scss 1:11  root stylesheet
+
+<===>
+================================================================================
+<===> error/known_incompatible/third/input.scss
+a {b: max(1px, 2px, 3s)}
+
+<===> error/known_incompatible/third/error
+Error: 1px and 3s are incompatible.
+  ,
+1 | a {b: max(1px, 2px, 3s)}
+  |           ^^^ 1px
+  |                     == 3s
+  '
+  input.scss 1:11  root stylesheet
+
+<===>
+================================================================================
+<===> error/potentially_incompatible_before_unitless/input.scss
+a {b: max(1c, 2d, 3)}
```

Comment:
I'm not sure I understand this case. How does this differ from `max(1d, 2, 3e)`, which doesn't error?

Reply:
It has to do with the order the numbers are compared against one another. For `max(1d, 2, 3e)` the algorithm first compares `1d` to `2`, which is valid because it's comparing a number with a unit to a number without, and then compares `2` to `3e`, which is valid for the same reason. Here though it first compares `1c` to `1d`, which is invalid because we don't know how to convert between those units. This is one of those bad cases that arises out of allowing unitless numbers to work with numbers with units.

Reply:
isn't the algorithm comparing a number to all the following ones in `SassCalculation._verifyCompatibleNumbers` ?

Reply:
In Dart Sass's implementation, `_verifyCompatibleNumbers` isn't called until after simplification occurs, and handling the semantics of `max()` is considered part of simplification.

Reply:
hmm, indeed. I forgot about that.